tools/sro2kmm/sro2kmm

```diff
@@ -0,0 +1,156 @@
+    exit 1
+fi
+
+# Check if --menu or -m is passed as an argument
+if [ -z "$2" ]; then
+    setdmenu="0"
+elif [[ $2 == "--menu" ]] || [[ $2 == "-m" ]]; then
+    setdmenu="1"
+else
+    echo "Not valid drain options. Use --menu or -m"
+    exit 1
+fi
+
+# Check if --verbose or -v is passed as an argument
+if [ -z "$3" ]; then
+    verbose=""
+elif [[ $3 == "--verbose" ]] || [[ $3 == "-v" ]]; then
+    verbose="-vvv"
+else
+    echo "Not valid verbose options. Use --verbose or -v"
+    exit 1
+fi
+
+# Set variables
+sr=$1
+ns=$(oc get sr $sr -o jsonpath='{.items[0]}{.spec.namespace}')
+echo "Namespace $ns"
+dsrawlist=$(oc get ds -n "$ns" -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
+echo "Dsrawlist: $dsrawlist"
+
+# Function to filter DaemonSets owned by SRO
+# Add only DaemonSets owned by SRO
+rm -rf data || true
+
+while read -r srods; do
+  tmpds=$(oc get ds $srods -n $ns -o json | jq --arg sro "$sr" 'try .metadata | select((.ownerReferences[].kind=="SpecialResource") and (.ownerReferences[].name==$sro)) | .name')
+  if [ -n "$tmpds" ]; then
+    mkdir -p data/$srods && echo $srods >> data/daemonset_list.txt
+  fi
+done <<< "$dsrawlist"
+
+# Create warning menu
+create_warning()
+{
+  dialog --msgbox "This script will reboot selected cluster nodes. You should identify what this process implies and take the additional measures that your systems may require. Access README.md for more details." 0 0
+}
+create_warning
+
+# Build dialog checklist box for possible DaemonSets
+dslist=$(cat data/daemonset_list.txt)
+COUNT=0
+while IFS=$'\n' read -r opt; do
+  COUNT=$(( COUNT+1 ))
+  options+=("$opt" "" off)
+done <<< "$dslist"
+
+cmd=(dialog --backtitle "SRO2KMM" --checklist "Choose one or more SRO DaemonSets to migrate" 22 60 16)
+
+#Add selected Daemonsets names to data file
+choices=($("${cmd[@]}" "${options[@]}" 2>data/selected_ds 1>/dev/tty))
+selected_ds=$(cat data/selected_ds)
+
+#Add selected DaemonSets namespace, names and containers to playbook variable
+add_sro_variables() {
+  rm -f vars/sro_ds_ns.yaml || true
+  echo "sro_ds_ns: $ns" > vars/sro_ds_ns.yaml
+  echo "sro_ds:" >> vars/sro_ds_ns.yaml
+  while read line|| [[ -n $line ]] ; do 
+	echo "   - $line" >> vars/sro_ds_ns.yaml 
+	oc get ds "$line" -n "$ns" -o 'jsonpath={range .spec.template.spec.containers[*]}{.name}{"\n"}{end}' >> data/${line}/containers.txt
+  done <<< "$selected_ds"
+}
+
+add_sro_variables
+# Create menus for each selected DaemonSet to make containers choice
+create_ds_containers_menu() {
+  while read -r ds; do
+    cns_ds_list=$(cat data/${ds}/containers.txt)
+    while read -r csn; do
+        optcsn+=("$csn" "" off)
+    done <<< "$cns_ds_list"
+    cmdcsn=(dialog --backtitle "SRO2KMM" --checklist "Choose one or more containers from DaemonSet $ds" 22 60 16)
+    choices=($("${cmdcsn[@]}" "${optcsn[@]}" 2>data/${ds}/containers.txt 1>/dev/tty))
+  done <<< "$selected_ds"
+}
+
+create_ds_containers_menu
```

Comment:
Unless I am missing something, these functions are called only once, just after their definition. Do we need them?